pkgs/by-name/vu/vue-language-server/package.nix

```diff
@@ -1,36 +1,73 @@
 {
   lib,
-  buildNpmPackage,
-  fetchurl,
+  stdenv,
+  fetchFromGitHub,
+  pnpm,
+  nodejs,
+  nix-update-script,
+  makeBinaryWrapper,
 }:
-
-buildNpmPackage rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "vue-language-server";
   version = "3.0.1";
 
-  src = fetchurl {
-    url = "https://registry.npmjs.org/@vue/language-server/-/language-server-${version}.tgz";
-    hash = "sha256-kSJgE94O1qYdd/YoRex8CJSprS15PpboTzChKnYj2nE=";
+  src = fetchFromGitHub {
+    owner = "vuejs";
+    repo = "language-tools";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-VPN4XWZDT44GwUZNih9sH2AiOKr8800B748DRoZ6hWc=";
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    fetcherVersion = 1;
+    hash = "sha256-X34wBcLHhCkpr6oFrUE33X8EjUjdSpV5TZQPYbMzRDs=";
   };
 
-  npmDepsHash = "sha256-S1kmTeReoAn428HaHQbwFzQg75rQNFVKVT1HT9m+7K4=";
-  forceGitDeps = true;
+  nativeBuildInputs = [
+    nodejs
+    pnpm.configHook
+    makeBinaryWrapper
+  ];
 
-  postPatch = ''
-    ln -s ${./package-lock.json} package-lock.json
+  buildPhase = ''
+    runHook preBuild
+    pnpm run build packages/language-server
+    runHook postBuild
   '';
 
-  dontNpmBuild = true;
-  makeCacheWritable = true;
+  preInstall = ''
+    pnpm prune --prod
+    find -type f \( -name "*.ts" -o -name "*.map" \) -exec rm -rf {} +
+
+    # https://github.com/pnpm/pnpm/issues/3645
+    find node_modules packages/language-server/node_modules -xtype l -delete
+
+    # remove non-deterministic files
+    rm node_modules/.modules.yaml
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib/language-tools}
+    cp -r {node_modules,packages,extensions} $out/lib/language-tools/
+
+    makeWrapper ${lib.getExe nodejs} $out/bin/vue-language-server \
+      --inherit-argv0 \
+      --add-flags $out/lib/language-tools/packages/language-server/bin/vue-language-server.js
+
+    runHook postInstall
+  '';
 
-  passthru.updateScript = ./update.sh;
+  passthru.updateScript = nix-update-script { };
 
-  meta = {
+  meta = with lib; {
```

Comment:
a nit. I think that the general standard is to remove as many `with` statements as possible and to be more explicit with dereference.

Reply:
Thx :), I've applied your suggestion.